frontend/simple/components/ProfilePicture.vue

```diff
@@ -0,0 +1,35 @@
+<template>
+ <img :src="picture">
+</template>
+<script>
+import {HapiNamespace} from '../js/backend/hapi'
+import {latestContractState} from '../js/state'
+var namespace = new HapiNamespace()
+export default{
+  name: 'ProfilePicture',
+  props: ['username'],
+  mounted: function () {
+    this.fetchData()
+  },
+  methods: {
+    fetchData: async function () {
+      if (this.username === this.$store.state.loggedIn) {
+        this.picture = this.$store.state[this.$store.getters.identity].attributes.picture
+      }
+      let identity = await namespace.lookup(this.username)
+      if (this.$store.state.contracts[identity]) {
+        this.picture = this.$store.state[identity].attributes.picture
+      } else {
+        let state = await latestContractState(identity)
+        this.picture = state.attributes.picture
+      }
+    }
+  },
+  data () {
+    return {
+      picture: ''
+    }
+  }
+}
+
```

Comment:
So this `fetchData` approach has always felt a bit off to me. There are a few issues with it:

- It's verbose
- It's kinda hackish
- It's not "Vue-like" in the sense that it doesn't react to updates in a clean way (and in the usage above, at all)

So here's my proposal for how to do this "the Vue / reactive way":

#### Principles

1. The usage should be as minimal as possible. This means, if a component isn't needed, don't make a component.
2. The state should always be available, and not need to be fetched. When it's updated by push notifications, the updates should spread reactively and not need to be fetched.

#### Implementation

**Option 1: No component**

If a component isn't needed, then the simplest solution (always the correct solution) is:

```html
<img :src="$store.state[userId].attributes.picture">
```

Boom. Done. No component needed and should update whenever the picture is updated. If that doesn't work, then use Vuex's getters-function functionality.

**Option 2: Component**

If a component is needed for some reason, that reason should be explicitly documented and justified in a comment at the top of the component.

Either way, the usage should be similar as above, with the non-component method, and there should be no `picture` data. The data should always be reactively fetched via Vuex getter functions.